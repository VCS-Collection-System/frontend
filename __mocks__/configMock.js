export const getConfig = (URL, configUrl, config) => {
  if(URL.indexOf(configUrl) > -1 ) {
    const country = URL.substring(configUrl.length);
    const countryConfig = config[country] || null;
    return Promise.resolve({data: countryConfig});
  } else {
    return Promise.reject({data: null});
  }
}