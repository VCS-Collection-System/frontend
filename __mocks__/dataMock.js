export const employeeInfo = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth":"2021-10-07",
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const attachmentObject = {
  "originalFileName":"blob",
  "size":100,
  "contentType":"image/png",
  "s3BucketName":"mockBucket",
  "s3UUID":"mockUUID"
}

export const vaxHistory = {
  "id":0,
  "employee": employeeInfo,
  "submissionDate":"2021-10-04T15:12:14.875",
  "review":{
     "reviewerEmployeeId":"z2",
     "reviewerNotes":"",
     "reviewDate":"2021-10-05T14:04:29.296",
     "reviewOutcome":"ACCEPTED",
     "rejectReason":""
  },
  "submittedBy":null,
  "autoApproved":false,
  "lastUpdatedTime":null,
  "lastUpdatedBy":null,
  "vaccineBrand":"MODERNA",
  "vaccineAdministrationDate":"2021-10-04",
  "fullVaccinatedFlag":false,
  "vaccineShotNumber":2,
  "documentType":"VaccineDocument",
  "attachment": {}
}

const taskEmployeeObject = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth": {
    "year": "2021",
    "monthValue": "12",
    "dayOfMonth": "16"
  },
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const mockTask = {
  "task-input-data": {
    "documentList": {
      "com.redhat.vcs.model.VaccineDocumentList": {
        "documents": [ {
          "com.redhat.vcs.model.VaccineCardDocument": {
            "employee": {
              "com.redhat.vcs.model.Employee": taskEmployeeObject
            },
            "vaccineBrand": {
              "com.redhat.vcs.model.VaccineBrand": "MODERNA"
            },
            "attachment": {
              "com.redhat.vcs.model.Attachment": attachmentObject
            },
            "vaccineAdministrationDate": {
              "year": "2021",
              "monthValue": "12",
              "dayOfMonth": "16"
            },
            "lotNumber": "mockLot"
          } }
        }
      }
    }
  }
}
