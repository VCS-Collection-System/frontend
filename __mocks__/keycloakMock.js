module.exports = {
    token: "bogustoken", 
    authenticated: false,
    loadUserProfile: jest.fn( ()=> {return Promise.resolve({
        attributes: {
            workforceid: ["mockId"]
        }
    })} ),
    login: jest.fn(),
    logout: jest.fn(),
    hasRealmRole: jest.fn(),
    profile: {
        firstName: "mockFirst",
        lastName: "mockLast",
        email: "mock@email.com",
        attributes: {
            workforceid: ["mockId"],
            agency: ["USA"]
        }
    }
};
