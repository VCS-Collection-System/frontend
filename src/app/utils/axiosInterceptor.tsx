import axios from "axios";

const axiosInterceptor = axios.create({})

axiosInterceptor.interceptors.request.use(
    config => {
        config.headers['Authorization'] = 'Bearer ' + sessionStorage.getItem("token");
        return config;
    },
    error => {
        Promise.reject(error)
    });
export default axiosInterceptor;