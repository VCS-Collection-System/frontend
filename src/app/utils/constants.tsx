declare global {  
  interface Window {
    REACT_APP_PAM_URL_BASE: string;
    REACT_APP_PAM_CONTAINER_ID: string;
    REACT_APP_KEYCLOAK_COUNTRY: string;
    REACT_APP_KEYCLOAK_EMAIL: string;
    REACT_APP_KEYCLOAK_FIRST_NAME: string;
    REACT_APP_KEYCLOAK_ID: string;
    REACT_APP_KEYCLOAK_LAST_NAME: string;
    REACT_APP_MAINTENANCE_BANNER: boolean;
    REACT_APP_MAINTENANCE_BANNER_TEXT: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_FEATURE_FLAGS: Array<Record<string, any>>;
    REACT_APP_MAX_DOSES: Record<string, number>;
    REACT_APP_CONSENT_LABEL: string;
    REACT_APP_ALTERNATE_CONSENT_LABEL: string;
    REACT_APP_BLOCKED_COUNTRIES: Array<string>;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_GLOBAL_CONFIG: Record<string, any>;
  }
}

// PAM urls
export const PAM_BASE_URL = window.REACT_APP_PAM_URL_BASE; 
export const SUBMIT_VAX_URL = PAM_BASE_URL + '/attestation/vax';
export const SUBMIT_TEST_URL = PAM_BASE_URL + '/attestation/covid-test-result';
export const PAM_TASK_URL = PAM_BASE_URL + '/tasks/';
export const RESERVED_TASK_URL = PAM_TASK_URL + 'owners?status=Reserved&user=';
export const INPROGRESS_TASK_URL = PAM_TASK_URL + 'owners?status=InProgress&user=';
export const ALL_TASK_URL = PAM_TASK_URL + 'owners?status=Ready&pageSize=200';
export const PRIORITY_URL = PAM_BASE_URL + '/query/priority-results';
export const PRIORITY_CONFIRM_URL = PAM_BASE_URL + '/query/priority-result/';
export const ATTACHMENT_URL = PAM_BASE_URL + '/attestation/attachment';
export const EMPLOYEE_URL = PAM_BASE_URL + '/query/employee/';
export const EMPLOYEE_NAME_URL = PAM_BASE_URL + '/query/employee-name/';
export const EMPLOYEE_VAX_HISTORY_URL = PAM_BASE_URL + '/query/vax-document/';
export const EMPLOYEE_TEST_HISTORY_URL = PAM_BASE_URL + '/query/test-result-document/';
export const VAX_TASK_DOCUMENT_URL = PAM_BASE_URL + '/query/task/document/';
export const DELETE_VAX_URL = PAM_BASE_URL + '/attestation/delete-vax-record/';
export const CONFIG_URL = PAM_BASE_URL + '/query/country-configs/';
export const SAVE_CONFIG_URL = PAM_BASE_URL + '/attestation/country-configs';

// List of countries allow to use this app
export const BLOCKED_COUNTRIES = window.REACT_APP_BLOCKED_COUNTRIES || []

export const KONAMI_CODE = [ "ArrowUp", "ArrowUp", "ArrowDown", "ArrowDown", "ArrowLeft", "ArrowRight", "ArrowLeft", "ArrowRight", "KeyB", "KeyA"];

// Maintenance Banner
export const MAINTENANCE_BANNER = window.REACT_APP_MAINTENANCE_BANNER || false
export const MAINTENANCE_BANNER_TEXT = window.REACT_APP_MAINTENANCE_BANNER_TEXT || ""

// Keycloak claims
export const KEYCLOAK_COUNTRY = window.REACT_APP_KEYCLOAK_COUNTRY
export const KEYCLOAK_EMAIL = window.REACT_APP_KEYCLOAK_EMAIL
export const KEYCLOAK_FIRST_NAME = window.REACT_APP_KEYCLOAK_FIRST_NAME
export const KEYCLOAK_ID = window.REACT_APP_KEYCLOAK_ID
export const KEYCLOAK_LAST_NAME = window.REACT_APP_KEYCLOAK_LAST_NAME
export const FEATURE_FLAGS = window.REACT_APP_FEATURE_FLAGS || []

// Maximum doses allowed by vaccine manufacturer
export const MAX_DOSES = window.REACT_APP_MAX_DOSES || {}

export const CONSENT_LABEL = window.REACT_APP_CONSENT_LABEL
export const ALTERNATE_CONSENT_LABEL = window.REACT_APP_ALTERNATE_CONSENT_LABEL

// Global country config
export const GLOBAL_CONFIG = window.REACT_APP_GLOBAL_CONFIG || {};

