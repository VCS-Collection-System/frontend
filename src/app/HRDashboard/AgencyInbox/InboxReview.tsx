import React, { useEffect, useState } from 'react';
import { 
  Button,
  Card, 
  CardBody, 
  CardFooter,
  Flex,
  FlexItem,
  Grid,
  GridItem,
  Split,
} from '@patternfly/react-core';
import { CheckCircleIcon, ErrorCircleOIcon, ExternalLinkAltIcon } from '@patternfly/react-icons';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useHistory } from 'react-router-dom';
import authaxios from "@app/utils/axiosInterceptor";
import ReviewInfoCard from '@app/Components/ReviewInfoCard';
import EmployeeInfoCard from '@app/Components/EmployeeInfoCard';
import { ResponseErrorModal } from '@app/Components/FeedbackModals';
import DeclineReasonModal from '@app/Components/DeclineReasonModal';
import { useKeycloak } from "@react-keycloak/web";
import { getEmployeeInfo } from '@app/utils/getEmployeeInfo';
import { PAM_TASK_URL, ATTACHMENT_URL } from "@app/utils/constants"

const PKG = "com.redhat.vcs.model.";
const VAX_DOCUMENT = PKG + "VaccineCardDocument";
const VAX_DOCUMENT_LIST = PKG + "VaccineDocumentList";
const VAX_BRAND = PKG + "VaccineBrand";
const VAX_EMPLOYEE = PKG + "Employee";
const VAX_ATTACHMENT = PKG + "Attachment";
const VAX_DOC_REVIEW = PKG + "DocumentReview";

const InboxReview: React.FunctionComponent = () => {
  const history = useHistory();
  const { keycloak } = useKeycloak();
  const employee = getEmployeeInfo(keycloak);
  const reviewerID = employee.id;

  const [isFormSubmitting, setIsFormSubmitting] = useState(false);
  const [isDeclineModalOpen, setIsDeclineModalOpen] = useState(false); 
  const [isErrorModalOpen, setIsErrorModalOpen] = useState(false);
  const [errorState, setErrorState] = useState(''); 
  const [doses, setDoses] = useState<Array<Record<string, string>>>([]);
  const [employeeId, setEmployeeId] = useState("");
  const [employeeName, setEmployeeName] = useState("");
  const [employeeDOB, setEmployeeDOB] = useState("");
  const [employeeEmail, setEmployeeEmail] = useState("");
  const [attachment, setAttachment] = useState("");
  const [attachmentType, setAttachmentType] = useState(""); 
  const [attachmentSize, setAttachmentSize] = useState(0); 

  useEffect(() => {
    const taskNumber = sessionStorage.getItem("approveId")
    const taskUrl = PAM_TASK_URL + taskNumber + "?withInputData=true&withOutputData=true&withAssignments=true"

    authaxios.get(taskUrl)
    .then(result => {
      let vaxDocument = null;
      if(result['data']['task-input-data']['document']) {
        vaxDocument = result['data']['task-input-data']['document'][VAX_DOCUMENT];

        if(vaxDocument != null) {
          const vaccineShotDate = vaxDocument['vaccineAdministrationDate'];
          const vaxDoses = [{
            vaccineBrand: vaxDocument['vaccineBrand'][VAX_BRAND],
            vaccineDate: vaccineShotDate['year'] + "-" + vaccineShotDate['monthValue'] + "-" + vaccineShotDate['dayOfMonth'],
            vaccineLotNumber: vaxDocument['lotNumber']
          }];
          setDoses(vaxDoses);
        }
      } else {
        const vaxDocuments = result['data']['task-input-data']['documentList'][VAX_DOCUMENT_LIST]['documents'];

        const vaxDoses = vaxDocuments.map((dose) => {
          dose = dose[VAX_DOCUMENT];
          const vaccineShotDate = dose['vaccineAdministrationDate'];
          return {
            vaccineBrand: dose['vaccineBrand'][VAX_BRAND],
            vaccineDate: vaccineShotDate['year'] + "-" + vaccineShotDate['monthValue'].toString().padStart(2, '0') + "-" + vaccineShotDate['dayOfMonth'].toString().padStart(2, '0'),
            vaccineLotNumber: dose['lotNumber']
          }
        });
        // Must sort because order of doses from server is not guaranteed to be in date order
        vaxDoses.sort((a, b) => {
          return a.vaccineDate.localeCompare(b.vaccineDate);
        });
        setDoses(vaxDoses);

        vaxDocument = vaxDocuments[vaxDocuments.length-1][VAX_DOCUMENT];
      }

      if(vaxDocument != null) {
        const employeeInfo = vaxDocument['employee'][VAX_EMPLOYEE]
        const dateOfBirth = employeeInfo['dateOfBirth']
        const resAttachment = vaxDocument['attachment'][VAX_ATTACHMENT]
        setAttachmentSize(resAttachment['size']);
        setEmployeeId(employeeInfo['employeeId'])
        setEmployeeName(employeeInfo['firstName'] + " " + employeeInfo['lastName'])
        if(dateOfBirth) {
          setEmployeeDOB(dateOfBirth['year'] + "-" + dateOfBirth['monthValue'] + "-" + dateOfBirth['dayOfMonth'])
        }
        setEmployeeEmail(employeeInfo['email']);

        authaxios.post(ATTACHMENT_URL, resAttachment, {
          responseType: 'arraybuffer'
        })
        .then(attachmentResult => {
          setAttachment(attachmentResult['data']);
          setAttachmentType(attachmentResult['headers']['content-type']);
        })
      }
    })
  }, []);

  const sendReview = (accepted, reason) => {
    setIsFormSubmitting(true)
    const taskNumber = sessionStorage.getItem("approveId")
    const taskUrl = PAM_TASK_URL + taskNumber + "/states/completed?auto-progress=true"
    const outcome = accepted ? "ACCEPTED" : "DECLINED"

    const acceptData = {
      "documentReview": {}
    }

    acceptData["documentReview"][VAX_DOC_REVIEW] = {
      "reviewerEmployeeId": reviewerID,
      "reviewOutcome": outcome,
      "reviewerNotes": reason,
      "rejectReason": reason
    }

    authaxios.put(taskUrl, acceptData, {
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      }
    })
    .then(() => {
      history.push("/hrdashboard")
    })
    .catch(res => {
      if(res.response) {
        setErrorState(res.response.status);
        setIsErrorModalOpen(true);
      }
    }).finally(()=> {
      setIsDeclineModalOpen(false)
      setIsFormSubmitting(false)
    })
  }

  const releaseTask = () => {
    const taskNumber = sessionStorage.getItem("approveId")
    const taskUrl = PAM_TASK_URL + taskNumber + "/states/released"
    const headers = {
      'Content-Type': 'application/json'
    }

    authaxios.put(taskUrl, {}, {
      headers: headers
    })
    .then(() => {
      history.push("/hrdashboard")
    })
    .catch(() => {
      history.push("/hrdashboard")
    })
  }

  return (
    <Card isRounded>
      <ResponseErrorModal 
        isModalOpen={isErrorModalOpen} 
        setIsModalOpen={setIsErrorModalOpen} 
        errorState={errorState}
      />
      <DeclineReasonModal 
        isModalOpen={isDeclineModalOpen}
        isSendingReview={isFormSubmitting}
        setIsModalOpen={setIsDeclineModalOpen}
        sendReview={sendReview}
      />
      <CardWithTitle title="HR Review" info="Review Vaccination Record"/>
      <CardBody>
        <Grid hasGutter>
          <GridItem rowSpan={2} span={6}> 
            <Flex justifyContent={{ default: "justifyContentCenter" }}>
              <FlexItem>
                {
                  attachment && attachmentSize === 0 &&
                  "No image"
                }
                {
                  attachment && attachmentType != 'application/pdf' && attachmentSize > 0 &&
                  <img src={URL.createObjectURL(new Blob([attachment]))} /> 
                }
                {
                  attachment && attachmentType == 'application/pdf' && attachmentSize > 0 &&
                  (
                    <Button
                      icon={<ExternalLinkAltIcon />}
                      variant="link"
                      component="a"
                      href={URL.createObjectURL(new Blob([attachment], {type: 'application/pdf'}))}
                      target="_blank"
                    >
                      Vaccination Card PDF
                    </Button>
                  )
                }
              </FlexItem>
            </Flex>
          </GridItem>
          <GridItem span={6}> 
            <ReviewInfoCard
              reviewType="VAX"
              doses={doses}
            />
            <EmployeeInfoCard 
              employeeId={employeeId}
              employeeName={employeeName}
              employeeDOB={employeeDOB}
              employeeEmail={employeeEmail}
            />
          </GridItem>
        </Grid>
      </CardBody>
      <CardFooter>
        <Split hasGutter>
          <Button 
            icon={<CheckCircleIcon />}
            isDisabled={isFormSubmitting}
            isLoading={isFormSubmitting}
            onClick={() => sendReview(true, "")}
            variant="primary" 
          >
            Accept
          </Button>
          <Button 
            icon={<ErrorCircleOIcon />} 
            variant="danger" 
            onClick = {() => setIsDeclineModalOpen(true)}
          >
            Decline
          </Button>
          <Button 
            variant="plain" 
            onClick={() => releaseTask()}
          >
            Back
          </Button>
        </Split>
      </CardFooter>
    </Card>
  );
}

export { InboxReview };
