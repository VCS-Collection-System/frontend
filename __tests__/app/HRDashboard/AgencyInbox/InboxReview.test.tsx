import React from 'react';
import { InboxReview } from '../../../../src/app/HRDashboard/AgencyInbox/InboxReview';
import mockKeycloak from '../../../../__mocks__/keycloakMock'
import { attachmentObject, mockTask } from '../../../../__mocks__/dataMock'
import { act } from 'react-dom/test-utils';
import { ATTACHMENT_URL,  PAM_TASK_URL } from '../../../../src/app/utils/constants'
import { fireEvent, render, screen } from '@testing-library/react'
import reactRouterDom from 'react-router-dom'

jest.mock('react-router-dom')
const mockPush = jest.fn()
reactRouterDom.useHistory = jest.fn().mockReturnValue({push: mockPush})

jest.mock('@react-keycloak/web', () => ({
  useKeycloak: jest.fn( ()=>({keycloak: mockKeycloak}))
})) 

jest.mock("@app/utils/constants", () => ({
  KEYCLOAK_FIRST_NAME: "firstName",
  KEYCLOAK_LAST_NAME: "lastName",
  KEYCLOAK_EMAIL: "email",
  KEYCLOAK_COUNTRY: "attributes.agency[0]",
  KEYCLOAK_ID: "attributes.workforceid[0]",
  APPROVER_ROLES: ["mock-approver"],
  FEATURE_FLAGS: ["fakeFeature"],
  VAX_TASK_DOCUMENT_URL: "vax_task_url/",
  PAM_TASK_URL: "pam_task_url/",
  GLOBAL_CONFIG: {"default": {
    "allowPdf": false,
    "dateOfBirth": true,
    "consentCheckbox": false,
    "covidTest": false,
    "customInput": true,
    "hrdashboard": true,
    "hrsearch": true,
    "proofType": {"cdc": false, "divoc": false, "greenPass": false},
    "recovery": false,
    "fakeFeature": true
  }}
}))

const pamURLEnding = "?withInputData=true&withOutputData=true&withAssignments=true"

jest.mock('@app/utils/axiosInterceptor', () => ({
  get: (URL) => {
    let returnData; 
    if(URL === PAM_TASK_URL + "mockID" + pamURLEnding){
      returnData = {data: mockTask}
    }
    if(URL === PAM_TASK_URL + "mockIDSix" + pamURLEnding){
      const pdfAttachment =  {
        "originalFileName":"mockPDF",
        "size":100,
        "contentType":"image/png",
        "s3BucketName":"mockBucket",
        "s3UUID":"mockUUID"
      }
      const pdfTask = mockTask;
      pdfTask['task-input-data']['documentList']['com.redhat.vcs.model.VaccineDocumentList']['documents'][0]['com.redhat.vcs.model.VaccineCardDocument']['attachment']['com.redhat.vcs.model.Attachment'] = pdfAttachment
      returnData = {data: pdfTask}
    }
    else{
      returnData = {data: mockTask};
    }
    return Promise.resolve(returnData)
  },
  post: (URL, props) => {
    let returnData; 

    if(URL === ATTACHMENT_URL && props.originalFileName === 'mockPDF'){
      returnData = {data: {}, headers: {"content-type": "application/pdf"}};
    }
    else if(URL === ATTACHMENT_URL){
      returnData = {data: {}, headers: {}};
    }
    return Promise.resolve(returnData)
  },
  put: (URL) => {
    let returnData; 
    const pamTaskEnding = "/states/completed?auto-progress=true"
    if(URL === PAM_TASK_URL + "mockID" + pamTaskEnding){
      returnData = {data: {}}
    }
    else if(URL === PAM_TASK_URL + "mockIDTwo" + pamTaskEnding){
      return Promise.reject({})
    }
    else if(URL === PAM_TASK_URL + "mockIDThree" + pamTaskEnding){
      return Promise.reject({response: {status: 500}})
    }
    else if(URL === PAM_TASK_URL + "mockIDFour" + "/states/released"){
      returnData = {data: {}}
    }
    else if(URL === PAM_TASK_URL + "mockIDFive" + "/states/released"){
      return Promise.reject({})
    }
    else{
      returnData = {data: {}};
    }
    return Promise.resolve(returnData)
  }
}))

beforeEach(() => {
  sessionStorage.setItem("approveId", "mockID");
  window.URL.createObjectURL = jest.fn();
})

afterEach(() => {
  sessionStorage.clear(); 
})

describe('<InboxReview />', ()=> {
  it('should be defined', () => {
    expect(InboxReview).toBeDefined()
  })

  it("should match existing snapshot", async () => {
    await act(async () => {
      const { asFragment } = render(
        <InboxReview />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  it("should display information from task", async () => {
    await act(async () => {
      render(<InboxReview />)
    })

    expect(screen.getByDisplayValue('MODERNA')).toBeTruthy();
    expect(screen.getByDisplayValue('mockLot')).toBeTruthy();
    expect(screen.getByDisplayValue('Mock Tester')).toBeTruthy()
  })

  it("should be able to accept a task", async () => {
    await act(async () => {
      render(<InboxReview />)
    })

    const acceptButton = screen.getByRole('button', {name: /Accept/i})

    await act(async () => {
      fireEvent.click(acceptButton)
    })

    expect(mockPush).toBeCalledTimes(1);
  })

  it("should be able to decline a task", async () => {
    await act(async () => {
      render(<InboxReview />)
    })

    const declineButton = screen.getByRole('button', {name: /Decline/i})

    await act(async () => {
      fireEvent.click(declineButton)
    })

    expect(screen.getByText('Attachments are not legible')).toBeTruthy();

    await act(async () => {
      fireEvent.click(screen.getByRole('radio', {name: /Attachments are not legible/i}))
      fireEvent.click(screen.getByRole('button', {name: /Confirm/i}))
    })

    expect(mockPush).toBeCalledTimes(1);
  })

  it("should react to submit error", async () => {
    sessionStorage.setItem("approveId", "mockIDTwo");
    await act(async () => {
      render(<InboxReview />)
    })

    const acceptButton = screen.getByRole('button', {name: /Accept/i})

    await act(async () => {
      fireEvent.click(acceptButton)
    })

    expect(mockPush).toBeCalledTimes(0);
  })

  it("should display network error", async () => {
    sessionStorage.setItem("approveId", "mockIDThree");
    await act(async () => {
      render(<InboxReview />)
    })

    const acceptButton = screen.getByRole('button', {name: /Accept/i})

    await act(async () => {
      fireEvent.click(acceptButton)
    })

    expect(screen.getByText('Server is unable to respond. Please try again later.')).toBeTruthy();
  })

  it("should return to hr dashboard when back button is clicked", async () => {
    sessionStorage.setItem("approveId", "mockIDFour");
    await act(async () => {
      render(<InboxReview />)
    })

    const backButton = screen.getByRole('button', {name: /Back/i})

    await act(async () => {
      fireEvent.click(backButton)
    })

    expect(mockPush).toBeCalledTimes(1)
    expect(mockPush).toBeCalledWith('/hrdashboard')
  })

  it("should return to hr dashboard even when network error", async () => {
    sessionStorage.setItem("approveId", "mockIDFive");
    await act(async () => {
      render(<InboxReview />)
    })

    const backButton = screen.getByRole('button', {name: /Back/i})

    await act(async () => {
      fireEvent.click(backButton)
    })

    expect(mockPush).toBeCalledTimes(1)
    expect(mockPush).toBeCalledWith('/hrdashboard')
  })

  it("should display PDFS", async () => {
    sessionStorage.setItem("approveId", "mockIDSix");
    await act(async () => {
      render(<InboxReview />)
    })

    expect(screen.getByText('Vaccination Card PDF')).toBeTruthy()
  })
})
