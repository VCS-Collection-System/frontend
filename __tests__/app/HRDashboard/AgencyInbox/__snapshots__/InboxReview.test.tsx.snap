// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InboxReview /> should match existing snapshot 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card pf-m-rounded"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <p
        class=""
        data-pf-content="true"
      >
        HR Review
      </p>
      <div
        class="pf-c-content"
      >
        <small
          class=""
          data-pf-content="true"
        >
          Review Vaccination Record
        </small>
      </div>
    </div>
    <div
      class="pf-c-card__body"
    >
      <div
        class="pf-l-grid pf-m-gutter"
      >
        <div
          class="pf-l-grid__item pf-m-6-col pf-m-2-row"
        >
          <div
            class="pf-l-flex pf-m-justify-content-center"
          >
            <div
              class=""
            />
          </div>
        </div>
        <div
          class="pf-l-grid__item pf-m-6-col"
        >
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-2"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__title"
            >
              <p
                class=""
                data-pf-content="true"
              >
                Review Information
              </p>
              <div
                class="pf-c-content"
              >
                <small
                  class=""
                  data-pf-content="true"
                />
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              <form
                class="pf-c-form"
                novalidate=""
              >
                <div
                  aria-label=""
                  class="pf-c-accordion pf-m-bordered"
                />
              </form>
            </div>
          </article>
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-3"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__title"
            >
              <p
                class=""
                data-pf-content="true"
              >
                Employee Information
              </p>
              <div
                class="pf-c-content"
              >
                <small
                  class=""
                  data-pf-content="true"
                />
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              <form
                class="pf-c-form"
                novalidate=""
              >
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="profile-form-employee-id"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Person ID
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="profile-input-employee-id"
                      class="pf-c-form-control"
                      disabled=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="profile-form-employee-name"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Name
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="profile-input-employee-name"
                      class="pf-c-form-control"
                      disabled=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="profile-form-employee-email"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Email
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="profile-input-employee-email"
                      class="pf-c-form-control"
                      disabled=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </form>
            </div>
          </article>
        </div>
      </div>
    </div>
    <div
      class="pf-c-card__footer"
    >
      <div
        class="pf-l-split pf-m-gutter"
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-primary pf-m-progress"
          data-ouia-component-id="OUIA-Generated-Button-primary-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          <span
            class="pf-c-button__icon pf-m-start"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              />
            </svg>
          </span>
          Accept
        </button>
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-danger"
          data-ouia-component-id="OUIA-Generated-Button-danger-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          <span
            class="pf-c-button__icon pf-m-start"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M409.5,319 C406.699326,316.168774 402.882413,314.575614 398.9,314.575614 C394.917587,314.575614 391.100674,316.168774 388.3,319 L319,388.3 C316.168774,391.100674 314.575614,394.917587 314.575614,398.9 C314.575614,402.882413 316.168774,406.699326 319,409.5 L421.5,512 L319,614.5 C316.168774,617.300674 314.575614,621.117587 314.575614,625.1 C314.575614,629.082413 316.168774,632.899326 319,635.7 L388.3,705 C391.100674,707.831226 394.917587,709.424386 398.9,709.424386 C402.882413,709.424386 406.699326,707.831226 409.5,705 L512,602.5 L614.5,705 C620.367618,710.877435 629.877455,710.922082 635.8,705.1 L705.1,635.8 C707.926508,632.970191 709.505822,629.128746 709.48721,625.129169 C709.468267,621.129592 707.852954,617.303145 705,614.5 L602.5,512 L705,409.5 C707.821142,406.720523 709.417879,402.93109 709.436648,398.97079 C709.455417,395.01049 707.89467,391.206092 705.1,388.4 L635.8,319 C629.932382,313.122565 620.422545,313.077918 614.5,318.9 L512,421.5 L409.5,319 Z M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 Z M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0 Z"
              />
            </svg>
          </span>
          Decline
        </button>
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-plain"
          data-ouia-component-id="OUIA-Generated-Button-plain-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          Back
        </button>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
